engine/execution/computation/computer/uploader/model.go

```diff
@@ -0,0 +1,50 @@
+package uploader
+
+import (
+	"io"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/onflow/flow-go/engine/execution"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/mempool/entity"
+)
+
+type BlockData struct {
+	Block       *flow.Block
+	Collections []*entity.CompleteCollection
+	TxResults   []*flow.TransactionResult
+	Events      []*flow.Event
+	TrieUpdates []*ledger.TrieUpdate
+}
+
+func ComputationResultToBlockData(computationResult *execution.ComputationResult) *BlockData {
```

Comment:
We only read the object, so I don't think we need any guarantees at this point. Assumption is that object is not modified while its being serialized.
The target format is something we agreed with DPS team

Reply:
> is this assumption actually guaranteed everywhere in the code?
No. Go in general lacks immutability of objects.

> And should we document this, so that people know that trying to perform the serialization while the object is being modified may have unexpected results?
Bigger part of our codebase (or any Go codebase really) would need such warning. This function does not modify the object, so if called is expecting modification it should create copy first.

Reply:
So in all cases where this function is being used, there would be no modification happening during or after the call, correct? In particular, there is no chance that `cr` will be further modified after this call:

https://github.com/onflow/flow-go/pull/1139/files#diff-9e005ebdb50b934004efede53802cbdda764e9c4090b5605dcf38d16a516b417R78

Reply:
No. We intent that serialization happens on an object which won't be modified. In EN this happens after the computation has happen